src/app/main/login/login.component.ts

```diff
@@ -27,15 +28,26 @@ export class LoginComponent implements OnInit {
   }
 
   login(){
+    const HTTP_OPTIONS = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/x-www-form-urlencoded',
+      })
+    };
+
+    const body = new HttpParams()
+      .set('username', this.form.value.username)
+      .set('password', this.form.value.password)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ticodacytest/chatApp1/pullRequest?prid=7784558)
```suggestion
      .set("password", this.form.value.password)
```